### PR TITLE
[ticket/15467] Fix JS for permissions setting

### PR DESCRIPTION
**Pull request by rxu**
Checklist:

- [x] Correct branch: master for new features; 3.2.x, 3.1.x for fixes
- [x] Tests pass
- [x] Code follows coding guidelines: [master / 3.2.x](https://area51.phpbb.com/docs/master/coding-guidelines.html), [3.1.x](https://area51.phpbb.com/docs/31x/coding-guidelines.html)
- [x] Commit follows commit message [format](https://wiki.phpbb.com/Git#Commit_Messages)

<a href="https://tracker.phpbb.com/browse/PHPBB3-15467">PHPBB3-15467</a>.

### CODE REVIEW

**Comment by rxu** [comment]
!set Styles

File: phpBB/adm/style/permission_mask.html [hunk 1]
```diff
@@ -9,6 +9,8 @@
 
 	var role_options = new Array();
 
+	var no_role_assigned = "{L_NO_ROLE_ASSIGNED}";
```
Comment (vinny) [inline phpBB/adm/style/permission_mask.html]: `LA_NO_ROLE_ASSIGNED` ?

Comment (rxu) [reply phpBB/adm/style/permission_mask.html]: Good catch.

File: phpBB/adm/style/permissions.js [hunk 2]
```diff
@@ -279,6 +279,16 @@ function reset_role(id) {
 	}
 
 	t.options[0].selected = true;
+
+	(function($)// Avoid conflicts with other libraries
```
Comment (hanakin) [inline phpBB/adm/style/permissions.js]: why are you using a sub function? just rewrite the whole function for jquery...Also please remove the conflict comments they are redundant

**Comment by rxu** [comment]
@hanakin Sure, the only minor issue is I have no idea how to do that.

File: phpBB/adm/style/permissions.js (hunk 2 above)
Comment (hanakin) [reply phpBB/adm/style/permissions.js]: 1. change `var t =` line to `var $t = $(id)`
2. `var parent = $(t).parent();` should just be `$t.parent();`
3. everywhere t is used change it to `$t`
4. wrap the whole reset_role function than in the self-invoking function.

Comment (rxu) [reply phpBB/adm/style/permissions.js]: @hanakin Thanks, but that kinda makes no sense for me to try blindly executing this instruction.
That simply leads me to facing errors like
```
permissions.js:271 Uncaught ReferenceError: jQuery is not defined
    at permissions.js:271
(anonymous) @ permissions.js:271
Uncaught ReferenceError: reset_role is not defined
    at HTMLAnchorElement.onclick 
Uncaught ReferenceError: reset_role is not defined
    at HTMLAnchorElement.onclick 
```

Where I have no clue about fixing them.
I guess it's easier to close the PR and leave it as is.

Comment (hanakin) [reply phpBB/adm/style/permissions.js]: hmm not sure why you did not get the error before then. Its saying that jquery is not loaded prior to the loading of permissions.js

**Comment by hanakin** [comment]
ah so yeah the permissions file is loaded in that specific template not sure why this is not in core.js probably should be

**Comment by rxu** [comment]
So what shall I do then? 😃

**Comment by hanakin** [comment]
I’ll investigate it tonight but I would say merge all that into the core.js file

**Comment by hanakin** [comment]
Sorry actually merge it into admin.js and that should solve everything

**Comment by rxu** [comment]
This all will lead to ```reset_role()``` becomes unworkable in ```onclick``` which requires complex changes in template as well as in js itself.
```
index.php?i=acp_permissions&sid=53512b9bf36564527e6ec9e4cdc6f54e&icat=16&mode=setting_group_local:257 Uncaught ReferenceError: reset_role is not defined
```
I'm not sure the trivial bug should trigger such a fundamental changes while it can be fixed by changing a couple of lines in the script. Neither I'm able to implement those, sorry.

**Comment by hanakin** [comment]
I think you misunderstood what I meant. permissions.js should not be included in a template this way. It should actually be merged into admin.js or moved to overaul_header.html via includejs. This is not really a bugs fault this is a global issue with all our js that eventually needs to be fixed. We actually need to rewrite all our js in the future for this and many other reasons, such as the bad practice in the way we are adding global vars in that file which should be converted to the data object but that's another issue. Just noticed you are trying to push this for 3.2 though. This might be outside the scope of 3.2 maybe avoid using jquery for 3.2? @marc1706 @Nicofuma ?

**Comment by marc1706** [comment]
I agree that this is a more broad issue. permissions.js should really be either merged into another file or included via INCLUDEJS. This probably will cause some other issues though. Not sure if it's worth it for 3.2.x, I'd much rather go for 3.3 and see if we can fix this in 3.2 without these major changes.

**Comment by rxu** [comment]
Can we postpone the decision with those weird "major changes" (which I just can't see) and simply fix this specific bug in 3.2/3.3?
The PR is updated.

**Comment by rxu** [comment]
> yes quick fix 3.2

Current PR state is right for that 😃

**Comment by marc1706** [comment]
BTW, will be included in 3.2.8.

**Pull request merged**